src/sentry/tasks/groupowner.py

```diff
@@ -77,7 +77,10 @@ def process_suspect_commits(event_id, event_platform, event_frames, group_id, pr
                     if created:
                         owner_count += 1
                         if owner_count > PREFERRED_GROUP_OWNERS:
-                            owners.first().delete()
+                            owner = owners.first()
+                            if owner:
+                                owner.delete()
+
```

Comment:
Can we just correctly do: 

```suggestion
                            try:
                                owner = owners[0]
                            except IndexError:
                                pass
                            else:
                                owner.delete()
```

Reply:
Okay, getting rid of first(). I changed owners[0] to owners.pop(0) (which also throws an IndexError if the index is out of bounds) because I want to remove the item from the list.